clientapp/app/components/shared/Measurements.tsx

```diff
@@ -1,22 +1,57 @@
-import React from 'react'
-interface IMeasurementsProps {
-  measurementRequest:string,
-  placeHolderExample:string,
+'use client'
+import React, { ChangeEvent, FormEvent } from 'react';
+import { useContext } from 'react';
+import { MeasurementContext,IMeasurements } from './context/MeasurementsData';
+import { sendMeasurementsToApi } from '@/app/Api/post_measurements';
 
-}
+export default function Measurements() {
+
+  const context = useContext(MeasurementContext);
```

Comment:
For the sake of easier test-writing, we may want to inject this context or it's associated functions as a dependencies to the Measurements component in the form of props. That allows us to test the calls to the `updateMeasurements` function very easily. Right now, it would be very hard to write a test for it otherwise.